crates/ruff_workspace/src/options.rs

```diff
@@ -377,7 +377,7 @@ pub struct Options {
 
     /// The lint sections specified at the top level.
     #[serde(flatten)]
-    pub lint_top_level: LintOptions,
+    pub lint_top_level: LintCommonOptions,
```

Comment:
The `LintCommonOptions` is necessary because we don't want to flatten the `exclude` option into the top-level settings (would  conflict with the global `exclude` setting)

Reply:
Can this be removed when we remove the top-level lint options support?

Reply:
I think that's the plan!

Reply:
I think we should probably wait a while to drop support for top-level lint options though, maybe 0.3? maybe later? We should open a tracking issue we can add todos to

Reply:
> Can this be removed when we remove the top-level lint options support?
Yes

> I think we should probably wait a while to drop support for top-level lint options though, maybe 0.3? maybe later? We should open a tracking issue we can add todos to

I don't know the timeline yet but my thinking is:

* Not before we ship the stable formatter (the main goal of the beta is to get feedback on the configuration)
* Promote the `lint` section over the `formatter` section (by changing the documentation)
* Deprecate the top-level settings
* Eventually, remove support for top-level lint settings

I expect this to be 0.3+